### DISCUSSION TITLE
@ConfigMapping not working in non-CDI context?

### DISCUSSION THREAD

**Discussion by edufolly** [quarkusio/quarkus#22421] [Q&A] [answered]
I'm trying to use `@ConfigMapping` in non-CDI context and get a null pointer exception with Quarkus 2.5.4.Final.

Everything works great in Quarkus 2.3.1.Final like https://github.com/quarkusio/quarkus/issues/20916

`application.properties`
```
server.host=https://myserver.com/client
```

`ServerConfig.java`
```java
@ConfigMapping(prefix = "server")
public interface ServerConfig {

    String host();

}
```

`ServerClient.java`
```java
@ApplicationScoped
public class ServerClient {

    public String get() {
        SmallRyeConfig config =
                ConfigProvider.getConfig().unwrap(SmallRyeConfig.class);

        ServerConfig server = config.getConfigMapping(ServerConfig.class);

        return server.host();
    }

}
```

`ServerResource.java`
```java
@Path("/api/v1/server")
public class ServerResource {

    @GET
    public String getServer() {
        return new ServerClient().get();
    }

}
```

Am I doing something wrong?

[a comment by an automated account is not shown]

**Reply by edufolly** [reply]
Ping @radcortez

**Reply by radcortez** [reply]
Ah, because the mapping is not used in any injection point it is removed. You need to add the `@Unremovable` annotation to your mapping.

**Reply by mkouba** [reply]
Also you should not use the `ServerClient` bean like this: `return new ServerClient().get();` but inject the bean in the `ServerResource` directly.

**Reply by edufolly** [reply]
Thanks @radcortez and @mkouba you are amazing! Both tips worked really well.

**Comment by edufolly** [comment, marked as answer]
Follow @mkouba answer:

`application.properties`
```
server.host=https://myserver.com/client
```

`ServerConfig.java`
```java
@ConfigMapping(prefix = "server")
public interface ServerConfig {

    String host();

}
```

`ServerClient.java`
```java
@ApplicationScoped
public class ServerClient {

    @Inject
    ServerConfig server;

    public String get() {
        return server.host();
    }

}
```

`ServerResource.java`
```java
@Path("/api/v1/server")
public class ServerResource {

    @Inject
    ServerClient client;

    @GET
    public String getServer() {
        return client.get();
    }

}
```
